src/ui/controls/edit.cpp

```diff
@@ -1940,7 +1936,7 @@ void CEdit::SetMaxChar(int max)
 
 int CEdit::GetMaxChar()
 {
-    return m_maxChar;
+    return m_text.capacity() - 1;
```

Comment:
m_text.capacity() can't really be used to determine the max amount of characters that CEdit is allowed to hold as it's value can be automatically adjusted by the STL implementation. This information should still be held as separate variable. See http://www.cplusplus.com/reference/string/string/capacity/

> Notice that this capacity does not suppose a limit on the length of the string. When this capacity is exhausted and more is needed, it is automatically expanded by the object (reallocating it storage space).
> 
> The capacity of a string can be altered any time the object is modified, even if this modification implies a reduction in size or if the capacity has not been exhausted (this is in contrast with the guarantees given to capacity in vector containers).